src/lib/fibers/sched.lua

```diff
@@ -88,12 +88,11 @@ function Scheduler:main()
 end
 
 function Scheduler:shutdown()
-   for i=1,100 do
+   repeat
```

Comment:
I see.  I am not sure about this change.  I agree that we will need to run the scheduler more iterations to drain it, but if something goes wrong while shutting down so that there's an infinite loop and it never finishes, the user should be told that something's wrong.

Can we instead run the scheduler while there are next tasks for up to a certain amount of time (maybe 0.1s), and if after that there are still runnable tasks, warn the user and proceed anyway?  It's not great of course, but maybe everything will just be garbage-collected later.

Besides that -- I think when shutting down we should neatly cancel the inotify streams.  More below.